certinject/cryptoapi_windows.go

```diff
@@ -0,0 +1,181 @@
+package certinject
+
+import "golang.org/x/sys/windows/registry"
+import "crypto/sha1"
+import "encoding/hex"
+import "strings"
+import "math"
+import "time"
+
+// In 64-bit Windows, this key is shared between 64-bit and 32-bit applications.
+// See https://msdn.microsoft.com/en-us/library/windows/desktop/aa384253.aspx
+const cryptoApiCertStoreRegistryBase = registry.LOCAL_MACHINE
+const cryptoApiCertStoreRegistryKey = `SOFTWARE\Microsoft\EnterpriseCertificates\Root\Certificates`
+const cryptoApiMagicName = "Namecoin"
+const cryptoApiMagicValue = 1
+
+func injectCertCryptoApi(derBytes []byte) {
+
+	// Format documentation of Microsoft's "Certificate Registry Blob":
+	
+	// 5c 00 00 00 // propid
+	// 01 00 00 00 // unknown
+	// 04 00 00 00 // size (little endian because Microsoft engineers suck at their jobs)
```

Comment:
Eh, little endian is fine. Windows is usually little endian and the adoption of big endian for network order was largely based on the fact that servers at the time (POWER, SPARC, etc.) were largely big endian.

Little endian has won (x86, ARM) and I'd probably use it for a binary network protocol if I were designing one today.